Web Server

AlexandrKras opened the pull request:
Добавлен веб-сервер к ДЗ про Hibernate ORM

AlexandrKras committed:
Web Server

stvort commented on HW24-webServer/build.gradle:
```diff
@@ -0,0 +1,34 @@
+    implementation("org.hibernate:hibernate-core")
+    implementation('org.flywaydb:flyway-core')
+
+    implementation("org.eclipse.jetty:jetty-server:${Versions.jetty}")
```
Проект не импортируется
1. HW24-webServer не подключен в sttings.gradle
2. В Versions нет версии jetty

stvort commented on HW24-webServer/src/main/java/ru/otus/hw/WebServerWithFilterBasedSecurityDemo.java:
```diff
@@ -0,0 +1,60 @@
+    private static final String TEMPLATES_DIR = "/web/templates/";
+
+    public static void main(String[] args) throws Exception {
+        MigrationsExecutor migrationsExecutor = new MigrationsExecutorFlyway(HIBERNATE_CFG_FILE);
+        migrationsExecutor.executeMigrations();
```
Вроде как в проекте нет никаких миграций, которые мог бы запустить `migrationsExecutor `

stvort commented on HW24-webServer/src/main/java/ru/otus/hw/WebServerWithFilterBasedSecurityDemo.java:
```diff
@@ -0,0 +1,60 @@
+        UserDao userDao = new UserDaoHibernate(sessionManager);
+        DBServiceUser dbServiceUser = new DbServiceUserImpl(userDao);
+
+        DataHelper.fillDb(dbServiceUser);
```
Хотя заполнение базы пользователями лучше бы сделать с помощью Repeatable миграций

stvort commented on HW24-webServer/src/main/resources/hibernate/hibernate.cfg.xml:
```diff
@@ -0,0 +1,21 @@
+
+    <property name="hibernate.show_sql">true</property>
+
+    <property name="hibernate.hbm2ddl.auto">create</property>
```
А саму схему создать версионными миграциями

stvort commented on HW24-webServer/src/main/java/ru/otus/hw/core/servlet/AdminPanel.java:
```diff
@@ -0,0 +1,83 @@
+
+import static javax.servlet.http.HttpServletResponse.SC_BAD_REQUEST;
+
+public class AdminPanel extends HttpServlet {
```
Лучше к названию добавить слово "Servlet"

stvort commented on HW24-webServer/src/main/java/ru/otus/hw/core/server/UsersWebServerSimple.java:
```diff
@@ -0,0 +1,75 @@
+
+        HandlerList handlers = new HandlerList();
+        handlers.addHandler(resourceHandler);
+        handlers.addHandler(applySecurity(servletContextHandler, "/users", "/api/user/*"));
```
Не увидел, чтобы на ` "/api/user/*"` был замапен сервлет

AlexandrKras committed:
1. Добавил миграцию базы (V1__initial_schema.sql)
2. Добавил заполнение базы через Repeatable миграций (R__create_data.sql)
3. Переиминовал класс "AdminPanel" по вашей рекомендации в "AdminPanelServlet"

AlexandrKras committed:
Merge branch 'master' into HW24-webServer
